### ISSUE TITLE
Flexibility for additional formats

### ISSUE THREAD

**Issue by ahoehn** [jormaechea/open-api-mocker#11]
According to https://swagger.io/docs/specification/data-models/data-types OAS3 supports much more formats then actually provided. Those for numbers are missing completely and for strings only 'password' is 'ignored'.
"format is an open value, so you can use any formats, even not those defined by the OpenAPI Specification" (spec OAS3), so a more open solution would be nice.

I succest to implement an additional parameter, were you can pass a in additional formats. This formats should be added as additional formats for ajv. I Propose the following config payload:
```
[{
  "type": "number" | "string",
  "name": "customformatname",
  "regex": "\\d*" //any js regex value is allowed to validate the custom format 
}]
```

in the schema validator additional formats can then be added to ajv with the following code:
```
additionalConfig.formats.map(format => 
	ajv.addFormat(format.name, {
		type: format.type,
		validate: (data => {
			const regex = RegExp(format.regex);
			const result = regex.test(data);
			logger.debug(`${format.name} check for ${data} results with ${result}`);
			return result;
		})
	})
);
```

**Comment by jormaechea** [comment]
Thanks for the contribution @ahoehn, I'll look into it as soon as I can.
By the way, I can saw that the PR has some failing checks. Please fix them before merging.

**Comment by jormaechea** [comment]
This was addressed and solved in #19
